descartes_core/include/descartes_core/path_planner_base.h

```diff
@@ -0,0 +1,53 @@
+/*
+ * path_planner_base.h
+ *
+ *  Created on: Jan 19, 2015
+ *      Author: ros developer 
+ */
+
+#ifndef DESCARTES_CORE_PATH_PLANNER_BASE_H_
+#define DESCARTES_CORE_PATH_PLANNER_BASE_H_
+
+#include <descartes_core/trajectory_pt.h>
+#include <descartes_core/robot_model.h>
+#include <vector>
+
+namespace descartes_core
+{
+
+class PathPlannerBase
+{
+public:
+  virtual ~PathPlannerBase(){}
+
+  /**
+   * @brief Plans a path for the given robot model through the points in the input trajectory.
+   * @param model robot model implementation for which to plan a path
+   * @param sampling This value will be interpreted differently by each planner specialization.
+   */
+  virtual bool initialize(RobotModelConstPtr &model, double sampling = 0.1f) = 0;
+
+  /**
+   * @brief Generates a robot path from the trajectory.
+   * @param traj the points used to plan the robot path
```

Comment:
One reason that 'getPath(...)' returns an array of generic TrajectoryPts is because the JointTrajectoryPt class has been moved out of descartes_core into the descartes_trajectory package.  
Furthermore, I agree that in most cases only joint values (JointTrajectoryPts) need to be returned, however future planner implementations may need to return additional data ( such as tool process data) or even point transition flags (linear,  joint move, tolerance zones) which could be applicable when translating these path points to robot programming language commands.  Thus, using the generic point type will allow returning point specialization for such cases.